builder/dockerfile/builder.go

```diff
@@ -199,7 +199,7 @@ func (b *Builder) build(ctx context.Context, source builder.Source, dockerfile *
 		targetIx, found := instructions.HasStage(stages, b.options.Target)
 		if !found {
 			buildsFailed.WithValues(metricsBuildTargetNotReachableError).Inc()
-			return nil, errdefs.InvalidParameter(errors.Errorf("failed to reach build target %s in Dockerfile", b.options.Target))
+			return nil, errdefs.InvalidParameter(errors.Errorf("failed to reach build target '%s' in Dockerfile", b.options.Target))
```

Comment:
```suggestion
			return nil, errdefs.InvalidParameter(errors.Errorf("failed to reach build target %q in Dockerfile", b.options.Target))
```

Reply:
I think this path is only for the classic builder? I just noticed that with BuildKit enabled we print a different error and wondering if that should also use the same approach? Or if we should align this error with the BuildKit one 🤔 


 e.g. I just tried;

```bash
docker build --target=nosuchtarget .
...
ERROR: failed to solve: target stage nosuchtarget could not be found
```

And without BuildKit;

```bash
DOCKER_BUILDKIT=0 docker build --target=nosuchtarget .
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.

Sending build context to Docker daemon  165.6MB
Error response from daemon: failed to reach build target nosuchtarget in Dockerfile
```

Reply:
Indeed, the buildkit message is on a vendor dependecy. IMHO the buildkit error should also quote the stage. What do you think ? I'd be happy to do a PR there too. Thanks !

Reply:
Yes, I was considering to update the error message here to be the same as Buildkit (but with the quotes); after that we can also update BuildKit to add quotes there

Reply:
Fixed ! Will make a PR in the buildkit repo 👍